cfme/automate/dialog_element.py

```diff
@@ -0,0 +1,199 @@
+    def __init__(self, parent):
+        self.parent = parent
+        Navigatable.__init__(self, appliance=parent.appliance)
+
+    @property
+    def tree_path(self):
+        return self.parent.tree_path
+
+    def instantiate(self, element_data=None):
+        return Element(self,
+            element_data=element_data)
+
+    def create(self, element_data=None):
+        for each_element in element_data:
+            view = navigate_to(self, "AddElement")
+            view.plus_btn.item_select("Add a new Element to this Box")
+            # View changes after clicking plus button
+            element_view = navigate_to(self, "Add")
+            element_view.fill(each_element)
+            self.element_type(element_view, each_element)
+        element_view.add_button.click()
+        element_view.flash.assert_no_error()
+        element_view.flash.assert_message('Dialog "{}" was added'.
+            format(self.parent.parent.parent.label))
+        element_view.flash.assert_no_error()
+        return self.instantiate(element_data=element_data)
+
+    def element_type(self, view, each_element):
+        """ Method to add element type.Depending on their type the subfields varies.
+
+        Args:
+            each_element: subfields depending on element type.
+        """
+        choose_type = each_element.get("choose_type")
+        dynamic_chkbox = each_element.get("dynamic_chkbox")
+        element_type = ['Drop Down List', 'Radio Button']
+        if choose_type in element_type:
+            if not dynamic_chkbox:
+                row = view.entry_table.row(Value='<New Entry>')
+                row.click()
+                view.fill({'entry_value': "Yes",
+                           'entry_description': "entry_desc"})
+                view.add_entry_button.click()
+            else:
+                node1 = "InspectMe"
+                view.fill({'field_entry_point': 'b'})
+                view.bt_tree.click_path("Datastore", "new_domain", "System", "Request", node1)
+                view.apply_btn.click()
+                view.fill({'field_show_refresh_button': True})
+        if choose_type == "Text Area Box":
+            view.fill({'text_area': 'Default text'})
+
+
+class Element(Navigatable):
+    """A class representing one Element of a dialog."""
+    def __init__(self, collection, element_data):
+        Navigatable.__init__(self, appliance=collection.appliance)
+        self.collection = collection
+        self.element_data = element_data
+
+    @property
+    def parent(self):
+        return self.collection.parent
+
+    @property
+    def element(self):
+        return ElementCollection(self)
+
+    @property
+    def tree_path(self):
+        return self.collection.tree_path
+
+    @property
+    def box(self):
+        return self.parent.box
+
+    def element_loc(self, element_data):
+        return sel.element('//div[@class="panel-heading"]'
+            '[contains(normalize-space(.), "{}")]/..'.format(element_data))
```

Comment:
Instead of:
```python
'some xpath ... "{}"'.format(foo)
```

please use
```python
from widgetastic.xpath import quote
'some xpath ... {}'.format(quote(foo))
```

Also please remove the `sel` reference and use self.browser instead.